webapp/channels/src/utils/notifications.ts

```diff
@@ -53,7 +53,7 @@ export async function showNotification(
 
     if (Notification.permission !== 'granted' && requestedNotificationPermission) {
         // User didn't allow notifications
-        return () => {};
+        return {result: 'not_sent', reason: 'notifications_permission_disallowed', data: Notification.permission, callback: () => {}};
```

Comment:
Instead of "disallowed", could this be something like `notifications_permission_previously_denied` so it's clearer when this happens?

Reply:
Sure, changed.